File: python/944.delete-columns-to-make-sorted.py
class Solution:
    def minDeletionSize(self, strs: list[str]) -> int:
        deleted = 0

        cols = len(strs[0])
        rows = len(strs)

        for c in range(cols):
            for r in range(1, rows):
                if strs[r - 1][c] > strs[r][c]:
                    deleted += 1
                    break

        return deleted

File: golang/944.delete-columns-to-make-sorted.go

func minDeletionSize(strs []string) int {
	rows := len(strs)
	cols := len(strs[0])

	deleted := 0

	for c := range cols {
		for r := 1; r < rows; r++ {
			if strs[r-1][c] > strs[r][c] {
				deleted++
				break
			}
		}
	}

	return deleted
}
